### PR TITLE
Add suppression case to CA1052

### PR DESCRIPTION
**Pull request by Youssef1313**
Related to https://github.com/dotnet/roslyn-analyzers/issues/3803

cc: @mavasani

### CODE REVIEW

**Comment by PRMerger12** [comment]
@Youssef1313 : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1052.md [hunk 1]
```diff
@@ -48,7 +48,10 @@ To fix a violation of this rule, mark the type as `static` and remove the defaul
 
 ## When to suppress warnings
 
-Suppress a warning from this rule only if the type is designed to be inherited. The absence of the `static` modifier suggests that the type is useful as a base type.
+Suppress a warning from this rule if:
+
+1. The type is designed to be inherited. The absence of the `static` modifier suggests that the type is useful as a base type.
+2. The type is used in as a type argument.
```
Comment (mavasani) [inline docs/code-quality/ca1052.md:54]:
```suggestion
It is fine to suppress violations in following cases:

1. The type is designed to be inherited. The absence of the `static` modifier suggests that the type is useful as a base type.
2. The type is designed to be used as a type argument. Static types cannot be used as type arguments.
```

**Comment by PRMerger8** [comment]
@Youssef1313 : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by Mikejo5000** [comment]
#sign-off

**Comment by PRMerger12** [comment]
@ktoliver : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Pull request merged**